src/website/shared/auth.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from typing import Any, cast
+
+from allauth.socialaccount.models import SocialAccount
+from django.contrib.auth.models import Group, Permission, User
+from django.db.models import Q
+from github import Github
+from github.NamedUser import NamedUser
+from github.Organization import Organization
+from github.Team import Team
+
+from shared.utils import get_gh
+
+github: Github = get_gh(per_page=100)  # 100 is the API limit
+logger = logging.getLogger(__name__)
+
+
+def get_gh_username(user: User) -> str | None:
+    """
+    Return the Github username of a given Auth.User.
+    """
+    social_user = User.objects.get(id=user.id)  # type: ignore
+    social_account: SocialAccount | None = (
+        social_user.socialaccount_set.filter(provider="github").first()  # type: ignore
+    )
+    if social_account:
+        return social_account.extra_data.get("login")  # type: ignore
+
+    logger.warning(f"Failed to get GitHub username for user {user}.")
```

Comment:
Please use `logger`'s interpolation as much as possible and do not resort to external formatting, otherwise the logger lose the opportunity to tap into the object you pass.

Reply:
Done [c698428](https://github.com/Nix-Security-WG/nix-security-tracker/pull/159/commits/c69842824f83be36e467f0d997f0afead9e0e01f)